src/formulas/tokenizer.ts

```diff
@@ -223,10 +213,50 @@ function tokenizeSpace(chars: string[]): Token | null {
   return null;
 }
 
-function tokenizeInvalidRange(chars: string[]): Token | null {
-  if (startsWith(chars, INCORRECT_RANGE_STRING)) {
-    chars.splice(0, INCORRECT_RANGE_STRING.length);
+function tokenizeInvalidRange(chars: TokenizingChars): Token | null {
+  if (chars.currentStartsWith(INCORRECT_RANGE_STRING)) {
+    chars.slice(INCORRECT_RANGE_STRING.length);
     return { type: "INVALID_REFERENCE", value: INCORRECT_RANGE_STRING };
   }
   return null;
 }
+
+class TokenizingChars {
+  private chars: string[];
+  private text: string;
+  private currentIndex: number = 0;
+
+  constructor(text: string) {
+    this.text = text;
+    this.chars = text.split("");
```

Comment:
do we actually need to split the string to an array? Can't we index the string directly?